### ISSUE TITLE
The Iou of Lane is 0

### ISSUE THREAD

**Issue by realTravisYou** [JiayuanWang-JW/YOLOv8-multi-task#89]
![image](https://github.com/user-attachments/assets/d3f31582-913b-4172-83b6-819b1d6f40b0)
I've trained over 100 epochs, but this IOU value is still 0. What is the problem? Thank you!
Here is my train file setting.
'import sys
sys.path.insert(0, "/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics")

from ultralytics import YOLO

model = YOLO('/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-n.yaml', task='multi')  # build a new model from YAML

model.train(data='/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=256, epochs=300, imgsz=640, device=(0,1,2,3), name='yolopm_represent', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)
'

**Comment by loganwu0526** [comment]
I have the same issue when the model is too small, but I already modify the original architecture and use my custom dataset

**Comment by realTravisYou** [comment]
> I have the same issue when the model is too small, but I already modify the original architecture and use my custom dataset

I want to reproduce the result of the paper, could you tell me how to modify the architecture to solve the problem? Thank you.

**Comment by JiayuanWang-JW** [comment]
> ![image](https://private-user-images.githubusercontent.com/143946752/376668314-d3f31582-913b-4172-83b6-819b1d6f40b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkzODgxMzgsIm5iZiI6MTcyOTM4NzgzOCwicGF0aCI6Ii8xNDM5NDY3NTIvMzc2NjY4MzE0LWQzZjMxNTgyLTkxM2ItNDE3Mi04M2I2LTgxOWIxZDZmNDBiMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMFQwMTMwMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMGM0YjdmNWY4OTI2MjRiZTBhZTM3NjFjMTEzNGIzMmYwZjI4MWEyNjQyOThlMzFiNDhiMGVjNDQwYzA0MGMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rob_cB4dbbX6VLW3Z-2weK2a_5FQYUZzRPm1pNAzAS0) I've trained over 100 epochs, but this IOU value is still 0. What is the problem? Thank you! Here is my train file setting. 'import sys sys.path.insert(0, "/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics")
> 
> from ultralytics import YOLO
> 
> model = YOLO('/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-n.yaml', task='multi') # build a new model from YAML
> 
> model.train(data='/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=256, epochs=300, imgsz=640, device=(0,1,2,3), name='yolopm_represent', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True) '

Hi, I think maybe the reason is your batch is too high. If you want to reproduce my result, try this training code. Please feedback if there are any issues, thanks. 

```
import sys
sys.path.insert(0, "/home/jiayuan/code/yolov8_v4/ultralytics")

from ultralytics import YOLO

# Load a model
model = YOLO('/home/jiayuan/code/yolov8_v4/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-m.yaml', task='multi')  # build a new model from YAML
# model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
# model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights

# Train the model
model.train(data='/home/jiayuan/code/yolov8_v4/ultralytics/datasets/bdd-multi.yaml', batch=54, epochs=300, imgsz=(640,640), device=[0,1,2], name='v4m-individual_640', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)
```

**Comment by realTravisYou** [comment]
> > ![image](https://private-user-images.githubusercontent.com/143946752/376668314-d3f31582-913b-4172-83b6-819b1d6f40b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkzODgxMzgsIm5iZiI6MTcyOTM4NzgzOCwicGF0aCI6Ii8xNDM5NDY3NTIvMzc2NjY4MzE0LWQzZjMxNTgyLTkxM2ItNDE3Mi04M2I2LTgxOWIxZDZmNDBiMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMFQwMTMwMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMGM0YjdmNWY4OTI2MjRiZTBhZTM3NjFjMTEzNGIzMmYwZjI4MWEyNjQyOThlMzFiNDhiMGVjNDQwYzA0MGMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rob_cB4dbbX6VLW3Z-2weK2a_5FQYUZzRPm1pNAzAS0) I've trained over 100 epochs, but this IOU value is still 0. What is the problem? Thank you! Here is my train file setting. 'import sys sys.path.insert(0, "/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics")
> > from ultralytics import YOLO
> > model = YOLO('/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-n.yaml', task='multi') # build a new model from YAML
> > model.train(data='/public/home/Travis/mnt/yt/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=256, epochs=300, imgsz=640, device=(0,1,2,3), name='yolopm_represent', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True) '
> 
> Hi, I think maybe the reason is your batch size is too large. If you want to reproduce my result, try this training code. Please feedback if there are any issues, thanks.
> 
> ```
> import sys
> sys.path.insert(0, "/home/jiayuan/code/yolov8_v4/ultralytics")
> 
> from ultralytics import YOLO
> 
> # Load a model
> model = YOLO('/home/jiayuan/code/yolov8_v4/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-m.yaml', task='multi')  # build a new model from YAML
> # model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
> # model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights
> 
> # Train the model
> model.train(data='/home/jiayuan/code/yolov8_v4/ultralytics/datasets/bdd-multi.yaml', batch=54, epochs=300, imgsz=(640,640), device=[0,1,2], name='v4m-individual_640', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)
> ```

Thank you for your kind reply, but I got the same result following your reply.
![image](https://github.com/user-attachments/assets/26510c3c-3361-4637-9aa6-46acba2a4755)
And my setting is 
```python
import sys
sys.path.insert(0, "/public/home/cszx_gwang/mnt/yt/YOLOv8-multi-task/ultralytics")

from ultralytics import YOLO

# Load a model
model = YOLO('/public/home/travis/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-n.yaml', task='multi')  # build a new model from YAML
# model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
# model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights

# Train the model
model.train(data='/public/home/travis/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=54, epochs=300, imgsz=640, device=[0,1,2], name='v4m-individual_640', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)

```

**Comment by JiayuanWang-JW** [comment]
This is interesting, this morning, I cloned this Repo to my server and training again. It works. I have trained 3 epochs for n scale model so far and I get the expected IoU for Lane. 
![image](https://github.com/user-attachments/assets/861a6e14-e132-48e0-8fb9-aa2fe1e86e0d)

My training code is as below: 

```
import sys
sys.path.insert(0, "/home/jiayuan/code/yolom/YOLOv8-multi-task/ultralytics")
# 现在就可以导入Yolo类了
from ultralytics import YOLO

# Load a model
model = YOLO('/home/jiayuan/code/yolom/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-n.yaml', task='multi')  # build a new model from YAML
# model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
# model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights

# Train the model
model.train(data='/home/jiayuan/code/yolom/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=36, epochs=300, imgsz=(640,640), device=[0,1], name='v4n-individual_640', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)
```

I only changed the number of GPUs and batch size because I only have 2 GPUs that can be used now. 

Could you provide more information about your running environment and please let me know how you installed the ultralytics: 'pip install -e .' or 'pip install ultralytics'?

**Comment by realTravisYou** [comment]
@JiayuanWang-JW Hi, I don't know why this works. I used the following code(the version of s)：
```python
import sys
sys.path.insert(0, "/public/home/travis/YOLOv8-multi-task/ultralytics")

from ultralytics import YOLO

# Load a model
model = YOLO('/public/home/travis/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-s.yaml', task='multi')  # build a new model from YAML
# model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
# model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights

# Train the model
model.train(data='/public/home/travis/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=54, epochs=300, imgsz=(640,640), device=[0,1,2], name='v4m-individual_640', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)
```
![image](https://github.com/user-attachments/assets/5c559be0-fc58-4768-af54-151f906539bf)
I am using CentOS, and my GPU model is A100 80G. The command I used was `pip install -e `. If I want to reproduce the results of a certain version of model m. Maybe I should try using a smaller batch size and two GPUs, as you suggested. Thank you for your patient response.

**Comment by JiayuanWang-JW** [comment]
> @JiayuanWang-JW Hi, I don't know why this works. I used the following code(the version of s)：
> 
> ```python
> import sys
> sys.path.insert(0, "/public/home/travis/YOLOv8-multi-task/ultralytics")
> 
> from ultralytics import YOLO
> 
> # Load a model
> model = YOLO('/public/home/travis/YOLOv8-multi-task/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-s.yaml', task='multi')  # build a new model from YAML
> # model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
> # model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights
> 
> # Train the model
> model.train(data='/public/home/travis/YOLOv8-multi-task/ultralytics/datasets/bdd-multi.yaml', batch=54, epochs=300, imgsz=(640,640), device=[0,1,2], name='v4m-individual_640', val=True, task='multi',classes=[2,3,4,9,10,11],combine_class=[2,3,4,9],single_cls=True)
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/143946752/378655632-5c559be0-fc58-4768-af54-151f906539bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NjUxMTksIm5iZiI6MTcyOTU2NDgxOSwicGF0aCI6Ii8xNDM5NDY3NTIvMzc4NjU1NjMyLTVjNTU5YmUwLWZjNTgtNDc2OC1hZjU0LTE1MWY5MDY1MzliZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMlQwMjQwMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kODNjZjRkNWIxY2VlY2RlN2ZjMTY3ZjAxODYyNmFiNWE5ZjI0ODBkNjg3ZGFkY2RiZGJjY2NjOTRjOTk1YmM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.H0Zxhjsa3Ix-qz_Vl9GfzJcqJJDdqVgt1nhBpElgtIw) I am using CentOS, and my GPU model is A100 80G. The command I used was `pip install -e `. If I want to reproduce the results of a certain version of model m. Maybe I should try using a smaller batch size and two GPUs, as you suggested. Thank you for your patient response.

@realTravisYou Thanks for your good news.

I compare your current code with the previous version. I think the difference is you change imgsz=(640,640) to imgsz=640. Maybe this is the issue. If you change other parts, please let me know. This will be helpful for other people. Thanks.

Actually, I use three 4090 GPUs with bs 120 to train the model for scale n as I mentioned in the paper. If you want to reproduce my results. Maybe you can follow my paper's parameters to train. If you have any issues, please reply to this issue and I will help you.

**Comment by realTravisYou** [comment]
@JiayuanWang-JW I figured out the cause of the problem, I annotate the following line of code as #22, so the IOU will always be 0.
![image](https://github.com/user-attachments/assets/274ad845-a7e2-4db6-8d09-7f99efe21090)
